src/contexts/CartContext/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { ReactNode, createContext, useEffect, useState } from 'react'
+
+import { CartContextType, CartItem } from './CartContext.types'
+
+export const CartContext = createContext<CartContextType>({
+  cart: [],
+  isOpen: false,
+  totalPrice: 0,
+  totalItems: 0,
+  toggleCart: () => undefined,
+  addItem: () => undefined,
+  updateItem: () => undefined,
+  deleteItem: () => undefined,
+  clearCart: () => undefined,
+})
+
+const CartContextProvider = ({ children }: { children: ReactNode }) => {
+  const storedCart = localStorage.getItem('cart')
+  const initialCart: CartItem[] = storedCart ? JSON.parse(storedCart) : []
+
+  const [cart, setCart] = useState<CartItem[]>(initialCart)
+
+  const [isOpen, setIsOpen] = useState(false)
+
+  const totalPrice = cart.reduce(
+    (acc, curr) => acc + curr.item.attributes.price * curr.quantity,
+    0,
+  )
+
+  const totalItems = cart.reduce((acc, curr) => acc + curr.quantity, 0)
+
+  useEffect(() => {
+    localStorage.setItem('cart', JSON.stringify(cart))
+  }, [cart])
+
+  const toggleCart = () => {
+    setIsOpen(!isOpen)
+  }
```

Comment:
I read about useCallback lately and now I'm wondering if I should use it on functions that don't depend on any variable (like toggleCart function). The thing is- that function is not 'expensive' in terms of performance, but on the other side it will be recreated every time CartContext will be updated. So the question is: is it necessary to wrap it around useCallback? I used useCallback in functions like addItem and updateItem, because calculations performed in them are longer comparing to other functions and they have 'cart' as a dependency- but what about other functions?